File: json-performance/performance-object-parser/src/main/java/org/jesperancinha/testing/hierarchy/Code.java
package org.jesperancinha.testing.hierarchy;

public class Code {

	protected String comment;
	private String name;

	protected ParentCode parent = null;

	public ParentCode getParent() {
		return parent;
	}

	public void setParent(ParentCode parent) {
		this.parent = parent;
	}

	public Code(String name, String comment) {
		this.setName(name);
		this.comment = comment;

	}

	public Code(ParentCode parent, String name) {
		this.setName(name);
		this.parent = parent;
	}

	protected String getName() {
		return name;
	}

	protected void setName(String name) {
		this.name = name;
	}

}

File: jeorg-kotlin-apps/performance-parsers/json-performance/performance-object-parser/src/main/kotlin/org/jesperancinha/testing/hierarchy/Code.kt
package org.jesperancinha.testing.hierarchy

open class Code {
    protected var comment: String? = null
    var name: String? = null
        protected set
    @JvmField
	var parent: ParentCode? = null

    constructor(name: String?, comment: String?) {
        this.name = name
        this.comment = comment
    }

    constructor(parent: ParentCode?, name: String?) {
        this.name = name
        this.parent = parent
    }
}